src/py-opentimelineio/opentimelineio/console/otiopluginfo.py

```diff
@@ -0,0 +1,225 @@
+    extra_features = []
+    for kind in ["read", "write"]:
+        if (
+            "{}_from_string".format(kind) in feature_map
+            and "{}_from_file".format(kind) not in feature_map
+        ):
+            extra_features.append(
+                "{0}_from_file (calls: {0}_from_string)".format(kind)
+            )
+
+    if extra_features:
+        print("    implicit supported features:")
+        for feat in extra_features:
+            print("      {}".format(feat))
+
+
+def _schemadefs_formatted(feature_map, args):
+    print("    SchemaDefs:")
+    for sd in feature_map.keys():
+        print("      {}".format(sd))
+        _docs_formatted(feature_map[sd]['doc'], args)
+
+
+def _docs_formatted(docstring, arg_map, indent=4):
+    long_docs = arg_map.get('long_docs')
+
+    if long_docs:
+        prefix = " " * indent + "doc (long): "
+    else:
+        prefix = " " * indent + "doc (short): "
+
+    initial_indent = prefix
+    subsequent_indent = " " * len(prefix)
+
+    block = docstring.split("\n")
+    fmt_block = []
+    for line in block:
+        line = textwrap.fill(
+            line,
+            initial_indent=initial_indent,
+            subsequent_indent=subsequent_indent,
+            width=len(subsequent_indent) + 80,
+        )
+        initial_indent = subsequent_indent
+        fmt_block.append(line)
+
+    if long_docs:
+        text = "\n".join(fmt_block)
+    else:
+        text = fmt_block[0]
+
+    print(text)
+
+
+_FORMATTER = {
+    "supported features": _supported_features_formatted,
+    "SchemaDefs": _schemadefs_formatted,
+    "doc": _docs_formatted,
+}
+
+_FIELDS_TO_SKIP = frozenset(["name"])
+
+
+def _print_field(key, val, **args):
+    # if attribute doesn't hit any of the user specified patterns
+    if (
+        not any(fnmatch.filter([key], pt) for pt in args['attribs'])
+        or key in _FIELDS_TO_SKIP
+    ):
+        return
+
+    if key in _FORMATTER:
+        _FORMATTER[key](val, args)
+        return
+
+    print("    {}: {}".format(key, val))
+
+
+def main():
+    """  main entry point  """
+    args = _parsed_args()
+
+    plugin_types = args.plugin_types
+
+    if 'all' in plugin_types:
+        # all means all but 'all'
+        plugin_types = OTIO_PLUGIN_TYPES[1:]
+
+    # load all the otio plugins
+    active_plugin_manifest = otio.plugins.ActiveManifest()
+
+    # list the loaded manifests
+    print("Manifests loaded:")
+    for mf in active_plugin_manifest.source_files:
+        print("  {}".format(mf))
+
+    for pt in plugin_types:
+        print("")
+        print("{}:".format(pt))
+        plugin_by_type = getattr(active_plugin_manifest, pt)
+        plugins = [
+            p for p in plugin_by_type
+            if fnmatch.filter([p.name], args.plugpattern)
+        ]
+        for plug in plugins:
+            print("  {}".format(plug.name))
+
+            if (
+                    pt in ["media_linkers"]
+                    and plug.name in os.environ.get(
```

Comment:
Any reason to not use `=` instead of `in` here? Also, would it be weird to call into media_linker.default_media_linker() for this to keep the logic centralized?

Reply:
good call